lib/regexp_parser/scanner/scanner.rl

```diff
@@ -877,12 +880,25 @@ module Regexp::Scanner
     @tokens
   end
 
+  # Emits an array with the details of the scanned pattern
```

Comment:
`#emit` had to be moved up because the private scope in 896 now applied to the methods below it (it did not apply to the old class methods)

Reply:
Makes sense and it looks better too.